File: server/model/tourist.js
const Sequelize = require('sequelize')
const MySequelize = require('../config/mysqlConfig')

const touristModel = MySequelize.define('tourist', {
  id: {
    type: Sequelize.STRING(19),
    field:  'id',
    primaryKey: true
  },
  nickName: {
    type: Sequelize.STRING(25),
    field:  'nickName',
    allowNull: false
  },
  telephone: {
    type: Sequelize.STRING(11),
    field:  'telephone',
    allowNull: false
  },
  password: {
    type: Sequelize.STRING(255),
    field:  'password',
    allowNull: false
  },
  email: {
    type: Sequelize.STRING(30),
    field:  'email',
    allowNull: false
  },
  birthday: {
    type: Sequelize.DATE,
    field:  'birthday',
    allowNull: false
  },
  gender: {
    type: Sequelize.STRING(1),
    field:  'gender',
    allowNull: false
  },
  address: {
    type: Sequelize.STRING(255),
    field:  'address',
    allowNull: false
  },
  avatar: {
    type: Sequelize.STRING(255),
    field:  'avatar',
    allowNull: true
  },
  is_deleted: {
    type: Sequelize.TINYINT,
    field:  'is_deleted',
    allowNull: false
  },
  create_time: {
    type: Sequelize.DATE,
    field:  'create_time',
    allowNull: false
  },
  modify_time: {
    type: Sequelize.DATE,
    field:  'modify_time',
    allowNull: false
  },
},
// {
//   tableName: 'tourist'   // 主要解决查询时将表名变复数的问题
// }
)

module.exports = touristModel

File: src/main/java/com/carolin_violet/travel_system/bean/Tourist.java
package com.carolin_violet.travel_system.bean;

import com.baomidou.mybatisplus.annotation.*;

import java.util.Date;
import java.io.Serializable;

import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.experimental.Accessors;

/**
 * <p>
 * 
 * </p>
 *
 * @author carolin-violet
 * @since 2022-05-30
 */
@Data
@EqualsAndHashCode(callSuper = false)
@Accessors(chain = true)
public class Tourist implements Serializable {

    private static final long serialVersionUID = 1L;

    @TableId(value = "id", type = IdType.ID_WORKER_STR)
    private String id;

    @TableField("nickName")
    private String nickName;

    private String avatar;

    private String telephone;

    private String password;

    private String email;

    private String gender;

    private String address;

    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern="yyyy-MM-dd", timezone = "GMT+8")
    private Date birthday;

    @TableLogic
    @TableField(fill = FieldFill.INSERT)
    private Integer isDeleted;

    @TableField(fill = FieldFill.INSERT)
    private Date createTime;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private Date modifyTime;


}
